storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -320,7 +324,10 @@ public Optional<String> serverConfigName(String configName) {
                 .define(TopicConfig.LOCAL_LOG_RETENTION_MS_CONFIG, LONG, DEFAULT_LOCAL_RETENTION_MS, atLeast(-2), MEDIUM,
                         TopicConfig.LOCAL_LOG_RETENTION_MS_DOC)
                 .define(TopicConfig.LOCAL_LOG_RETENTION_BYTES_CONFIG, LONG, DEFAULT_LOCAL_RETENTION_BYTES, atLeast(-2), MEDIUM,
-                        TopicConfig.LOCAL_LOG_RETENTION_BYTES_DOC);
+                        TopicConfig.LOCAL_LOG_RETENTION_BYTES_DOC)
+                .define(TopicConfig.REMOTE_LOG_DISABLE_POLICY_CONFIG, STRING, TopicConfig.REMOTE_LOG_DISABLE_POLICY_RETAIN,
```

Comment:
Do we plan to finish implementing this feature in 3.9, which is a short release? If not, perhaps we should define the new config with `defineInternal` so that it's not exposed to the end user until the feature is implemented.